pkgs/by-name/ce/certmonger/package.nix

```diff
@@ -0,0 +1,151 @@
+{
+  lib
+  , stdenv
+  , fetchurl
+  , automake
+  , autoconf
+  , gettext
+  , pkg-config
+  , writeShellScript
+  , openldap
+  , libkrb5
+  , dbus
+  , nspr
+  , nss
+  , openssl
+  , libidn2
+  , libuuid
+  , talloc
+  , tevent
+  , curl
+  , libxml2
+  , xmlrpc_c
+  , jansson
+  , popt
+  , coreutils
+  , runtimeDir ? "/run/certmonger"
+  , storeDir ? "/var/lib/certmonger"
+}:
+let
+  # patch runtime /libexec + storeDir (it is hardcoded in binary for runtime)
+  # needed for non-NixOS systems
+  # For example "certmonger" will create file in "/var/lib/certmonger/cas/20240222033838" with hardcoded path:
+  # ca_external_helper=/nix/store/h3xp6zjjvq5gy2wcllfv5l578fbjr9il-certmonger-0.79.15/libexec/certmonger/certmaster-submit
+  certmongerWrapper = writeShellScript "certmongerWrapper" ''
+    # create runtime impure binaries
+    if [ ! -d "${runtimeDir}" ]; then
+      ${coreutils}/bin/mkdir -m 0750 -p "${runtimeDir}"
+    fi
+    if [ "$(${coreutils}/bin/readlink "${runtimeDir}/libexec")" != "@out@/libexec" ]; then
+        ${coreutils}/bin/rm -f "${runtimeDir}/libexec"
+        ${coreutils}/bin/ln -s "@out@/libexec" "${runtimeDir}/libexec"
+    fi
+
+    # create required directories
+    if [ ! -d "${storeDir}/cas" ]; then
+      ${coreutils}/bin/mkdir -m 0700 -p "${storeDir}/cas"
+    fi
+    if [ ! -d "${storeDir}/local" ]; then
+      ${coreutils}/bin/mkdir -m 0700 -p "${storeDir}/local"
+    fi
+    if [ ! -d "${storeDir}/requests" ]; then
+      ${coreutils}/bin/mkdir -m 0700 -p "${storeDir}/requests"
```

Comment:
I don't think we need `${coreutils}` here? Have you tried doing just eg. mkdir?

Reply:
fixed, coreutils is already in path. Thanks!